obsreport/obsreport_exporter.go

```diff
@@ -64,19 +64,14 @@ type ExporterSettings struct {
 	ExporterCreateSettings component.ExporterCreateSettings
 }
 
-// Deprecated: [v0.64.0] use MustNewExporter.
-func NewExporter(cfg ExporterSettings) *Exporter {
-	exp, err := newExporter(cfg, featuregate.GetRegistry())
-	if err != nil && cfg.ExporterCreateSettings.Logger != nil {
-		cfg.ExporterCreateSettings.Logger.Warn("Error creating an obsreport.Exporter", zap.Error(err))
-	}
-
-	return exp
+// NewExporter creates a new Exporter.
+func NewExporter(cfg ExporterSettings) (*Exporter, error) {
+	return newExporter(cfg, featuregate.GetRegistry())
```

Comment:
Don't need the private one. Just rename newExporter -> NewExporter

Reply:
The private one is mostly to wrap the `featuregate` since it's only used for tests.